src/source/lexstring.d

```diff
@@ -195,16 +195,17 @@ mixin template LexStringImpl(Token,
 	}
 
 	EscapeSequence lexOctalEscapeSequence(uint begin) {
-		uint r = 0;
-		foreach (i; 0 .. 3) {
+		uint r = frontChar - '0';
```

Comment:
By doing this, you are adding the assumption that the first character is a number. But you don't know that here.

This require a precondition, or we can just leave the loop as it was (in general, compilers are pretty good at unrolling, so I don't think it's a major problem).

I'm fine with either solution. Other than that, all good!

Reply:
Well, it is called from `lexEscapeSequence`, which already checks that. Though, adding an assert is a good idea.

Reply:
If this is expected from the caller, then it should be an in contract.

Reply:
I added an assert

Reply:
Or, you mean D contract. Will do

Reply:
done